charts/datadog-operator/templates/_helpers.tpl

```diff
@@ -52,20 +52,89 @@ Create the name of the service account to use
 {{- end -}}
 
 {{/*
-Return secret name to be used based on provided values.
+Return the value for a given data key in the datadog endpoint-config ConfigMap.
+*/}}
+{{- define "get-endpoint-config-data-key" -}}
+{{- $ctx := index . 0 }}
+{{- $key := index . 1 }}
+{{- $ns := $ctx.Release.Namespace -}}
+{{- $list := lookup "v1" "ConfigMap" $ns "" -}}
+{{- if $list }}
+  {{- range $list.items }}
+    {{- if hasSuffix "endpoint-config" .metadata.name  }}
+      {{- get .data $key -}}
+    {{- end }}
+  {{- end }}
+{{- end }}
+{{- end -}}
+
+{{/*
+Return true if value for a given key in the datadog endpoint-config ConfigMap is valid.
+*/}}
+{{- define "is-valid-endpoint-config-data" -}}
+{{- $ctx := index . 0 }}
+{{- $key := index . 1 }}
+{{- $val := include "get-endpoint-config-data-key" (list $ctx $key) -}}
+{{- if gt (len $val) 0 -}}
+true
+{{- else -}}
+false
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if DD_API_KEY env var should be set.
+*/}}
+{{- define "should-set-dd-api-key" -}}
+{{- if or .Values.apiKey .Values.apiKeyExistingSecret (eq (include "is-valid-endpoint-config-data" ( list . "api-key-secret-name")) "true") -}}
+true
+{{- else -}}
+false
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if DD_APP_KEY env var should be set.
+*/}}
+{{- define "should-set-dd-app-key" -}}
+{{- if or .Values.appKey .Values.appKeyExistingSecret (eq (include "is-valid-endpoint-config-data" ( list . "app-key-secret-name")) "true") -}}
+true
+{{- else -}}
+false
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return apiKey secret name to be used based on provided values.
+Priority for determining secret name:
+1. .Values.apiKey
+2. .Values.apiKeyExistingSecret
+3. api-key-secret-name from endpoint-config configMap
 */}}
 {{- define "datadog-operator.apiKeySecretName" -}}
+{{- if and (eq (include "is-valid-endpoint-config-data" (list . "api-key-secret-name")) "true") (not .Values.apiKey) (not .Values.apiKeyExistingSecret) }}
+{{- (include "get-endpoint-config-data-key" (list . "api-key-secret-name")) }}
+{{- else }}
 {{- $fullName := printf "%s-apikey" (include "datadog-operator.fullname" .) -}}
 {{- default $fullName .Values.apiKeyExistingSecret | quote -}}
```

Comment:
```suggestion
{{- default $fullName .Values.apiKeyExistingSecret -}}
```
nit: for consistency, remove `| quote` from the else block to match the unquoted format in the if block

Reply:
I think we should keep this the same as before since this line is what will be rendered by Helm. 

The `.Values.apiKeyExistingSecret` statement in the `if` block is only being evaluated for the conditional check